### ISSUE TITLE
logout with OIDC does not delete local storage database

### ISSUE THREAD

**Issue by jonathanmmm** [pulsejet/nextcloud-oidc-login#115]
Hi, 

I am using riotchat/element app and when I am logging in via this app it doesn't delete this database:
https://github.com/gary-kim/riotchat/issues/380
this means that if another user logs in again he is still logged into riotchat/element (even while logging out the login session SSO key is deleted when logging out, deleting this database in the browser under cookies manually, solves this problem).

Would be really great to get help, I am giving every info, debug that somebody could need, can even provide testing accounts.

This is on shared computers a great security risk. Using LDAP or normal login and using OIDC only in riotchat/element works.
You can see, what we tried in the other issue.
So somehow a problem is with these both apps at the same time.

I am using keycloak as auth provider and openldap as LDAP.

This is my config in config.php, I am using the official nextcloud-fpm docker image behind nginx proxy, so apps are stored into the folder custom_apps.
```
 'oidc_login_provider_url' => 'https://keycloak.my-domain.com/auth/realms/my-domain',
  'oidc_login_client_id' => 'nextcloud-openid-connect',
  'oidc_login_client_secret' => 'somestring',
  'oidc_login_logout_url' => 'https://keycloak.my-domain.com/auth/realms/my-domain/protocol/openid-connect/logout?redirect_uri=https%3A%2F%2Fnextcloud.my-domain.com',
  'oidc_login_hide_password_form' => true,
  'oidc_login_proxy_ldap' => true,
  'oidc_login_auto_redirect' => false,
  'oidc_login_button_text' => 'OpenID Connect Login',
  'oidc_login_attributes' =>
  array (
    'id' => 'sub',
    'name' => '${given_name} ${family_name}',
    'mail' => 'email',
    'ldap_uid' => 'preferred_username',
  ),
  'oidc_login_scope' => 'openid profile',
  'oidc_login_disable_registration' => true,

```

**Comment by jonathanmmm** [comment]
Ok, I got the error down to 'oidc_login_logout_url'
if I comment it out, I get logged out of nextcloud (database deleted), but not logged out of keycloak, so clicking openid connect login logs me in without asking for credentials. But I am no longer logged into element.

I tried in Keycloak with
`Backchannel Logout URL `

this is the description:
`URL that will cause the client to log itself out when a logout request is sent to this realm (via end_session_endpoint). If omitted, no logout request will be sent to the client is this case.
`

I put `https://nextcloud.my-domain.com/logout` in it, but it didn't do either.

So the logout from nextcloud doesn't work correctly or the one of keycloak (openid connect)

**Comment by jonathanmmm** [comment]
@pulsejet can I give anymore info on how to solve this problem?

I created a 15$ bounty on https://www.bountysource.com/issues/100783086-logout-with-oidc-does-not-delete-local-storage-database

**Comment by Mvmo** [comment]
Can you contact me privately. Either on discord or email

Discord: mvmo#9595
E-Mail: mvmogo@gmail.com

**Comment by jonathanmmm** [comment]
> Can you contact me privately. Either on discord or email
> 
> Discord: mvmo#9595 E-Mail: [mvmogo@gmail.com](mailto:mvmogo@gmail.com)

@Mvmo Contacted you via E-Mail, couldn't find you on Discord with your name#tag

**Comment by flo-mic** [comment]
@jonathanmmm  was you able to solve this?

**Comment by jonathanmmm** [comment]
@flo-mic no, I couldn't solve it.

**Comment by sirkrypt0** [comment]
Hey,
thanks for bringing this up again. This is indeed an issue on our site and #164 should fix this.

**Comment by jonathanmmm** [comment]
@sirkrypt0 thanks, I have not reviewed till don't know, probably when I opened the issue (tried it right now and still there, but manually fixed the file like in the merge and the issue is gone). When this fix is released in the store and working I will give you (or who else did it) the bounty. Thanks :-)

**Comment by sirkrypt0** [comment]
> @sirkrypt0 thanks, I have not reviewed till don't know, probably when I opened the issue (tried it right now and still there, but manually fixed the file like in the merge and the issue is gone). When this fix is released in the store and working I will give you (or who else did it) the bounty. Thanks :-)

The fix was just released yesterday in v2.3.1, so check it out and let me know if it works as expected :)

**Comment by jonathanmmm** [comment]
@sirkrypt0 works, so I will close this. As you seem to be the one having done the work and the pull request, you can claim the bounty. Thanks again :-)
